pybandits/strategy.py

```diff
@@ -20,15 +20,61 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+from abc import ABC, abstractmethod
 from random import random
-from typing import Dict, List, Optional
+from typing import Any, Dict, List, Optional, TypeVar, Union
 
 import numpy as np
-from pydantic import validate_call
+from pydantic import field_validator, validate_call
 from scipy.stats import ttest_ind_from_stats
 
-from pybandits.base import ActionId, Float01, Model, Probability, Strategy
-from pybandits.model import Beta, BetaMOCC
+from pybandits.base import ActionId, Float01, Probability, PyBanditsBaseModel
+from pybandits.model import Beta, BetaMOCC, Model
+
+_Self = TypeVar("_Self", bound="Strategy")
```

Comment:
@validate_call on the "with_*" methods blocks standard type hinting methods, such as "ClassName" or using ClassName in conjunction with `from __future __ import annotations`. So, in order to maintain compatability with python 3.8-3.11, this "manaul" solution is used. With python 3.11, one can simply use `from typing import Self`.